framework/src/mesh/MooseMesh.C

```diff
@@ -1000,10 +1000,57 @@ MooseMesh::getBoundaryElementRange()
 
 const std::unordered_map<boundary_id_type, std::unordered_set<dof_id_type>> &
 MooseMesh::getBoundariesToElems() const
+{
+  mooseDeprecated("MooseMesh::getBoundariesToElems is deprecated, "
+                  "use MooseMesh::getBoundariesToActiveLocalElemIds");
+  return getBoundariesToActiveLocalElemIds();
+}
+
+const std::unordered_map<boundary_id_type, std::unordered_set<dof_id_type>> &
+MooseMesh::getBoundariesToActiveLocalElemIds() const
 {
   return _bnd_elem_ids;
 }
 
+const std::unordered_set<dof_id_type>
+MooseMesh::getBoundaryActiveLocalElemIds(BoundaryID bid) const
+{
+  // The boundary to element map is computed on every mesh update
+  const auto it = _bnd_elem_ids.find(bid);
+  if (it == _bnd_elem_ids.end())
+    // Boundary is not local to this domain, return an empty set
+    return std::unordered_set<dof_id_type>{};
+  return it->second;
+}
+
+const std::unordered_set<dof_id_type>
+MooseMesh::getBoundaryActiveNeighborElemIds(BoundaryID bid) const
+{
+  // Vector of boundary elems is updated every mesh update
+  std::unordered_set<dof_id_type> neighbor_elems;
+  for (const auto & local_elem : _bnd_elems)
```

Comment:
I think we should be able to use a structured binding for the element, side, and boundary ID here. I believe this matches Case 3 here: https://en.cppreference.com/w/cpp/language/structured_binding

Reply:
ok I tried